### DISCUSSION TITLE
Allow admins to order relationship fields with drag-and-drop

### DISCUSSION THREAD

**Discussion by Snailedlt** [payloadcms/payload#4068] [Feature Requests & Ideas]
## What?
Currently there's no way for admins to order the options in a hasMany relationship field. It would be nice if they could order the relationships by for example drag-and-drop. Perhaps ordering could be a new attribute `orderable`, `admin.orderable` or `admin.dragAndDropOrdering` which is just a boolean?

## Why?
This might be useful for things like tags, where you'd want, say 10 tags... but in the frontend you'd only want to display 4 of them. The admin would like to control which 4 are displayed, so having them ordered by priority would be an easy way to allow that.

**Comment by jacobsfletch** [comment]
@Snailedlt can you achieve this using `filterOptions`? I'm not sure if it allows you to set a `sort` but feels like that's the way to achieve this. You could simply sort on a priority field.

**Reply by Snailedlt** [reply]
Possibly. I haven't looked into filterOptions yet.


Do you have some example code in mind?

**Reply by linobino1** [reply]
`filterOptions` only allows for a `Where` condition. There is also `sortOptions`, but that only allows for sorting by a field of the related collection I think.

**Comment by dancarlson** [comment]
This is a standard feature in any CMS, hope it is added asap;  now that v3 is rockin. Drupal has a button to reset to alphabetical which is bonus. gg

**Comment by Snailedlt** [comment]
I forgot to mention, but the current workaround that I'm using is to have an array wrapping the relationship field, like so:

```ts
{
	name: "tags",
	labels: {
		singular: "Tag",
		plural: "Tags",
	},
	type: "array",
	maxRows: 10,
	fields: [
		{
			name: "tag",
			type: "relationship",
			relationTo: Tag.slug,
			required: true,
		},
	],
},
```

Not ideal, especially because it adds another layer of depth, but it works for now.

**Reply by dancarlson** [reply]
Excellent Snailedit! arrays are sortable; i assume that loading them default sort respects this sort order? (redacted)

**Reply by Snailedlt** [reply]
@dancarlson iirc, yes. The sort order is respected when fetched by the REST API at least

**Comment by DmacMcgreg** [comment]
This is a very much nice to have.
